tests/UI/pages/BO/catalog/discounts/add.js

```diff
@@ -257,9 +257,15 @@ class AddCartRule extends BOBasePage {
     // Fill actions form
     await this.fillActionsForm(page, cartRuleData);
 
-    // Save and return successful message
-    await this.clickAndWaitForNavigation(page, this.saveButton);
-    return this.getAlertSuccessBlockContent(page);
+    if (waitForNavigation) {
+      // Save and return successful message
+      await this.clickAndWaitForNavigation(page, this.saveButton);
+      return this.getAlertSuccessBlockContent(page);
+    }
+
+    // Save
+    await this.waitForSelectorAndClick(page, this.saveButton);
+    return page;
```

Comment:
```suggestion
    return null;
```

No need to return page, i think